### PR TITLE
[NAN-177] Add in activity log filtering

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Allow logs to be filtered upon by status, script, integration, connection, and/or date

## Issue ticket number and link
NAN-177

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/activity/activity.service.ts [hunk 1]
```diff
@@ -205,6 +222,36 @@ export async function getTopLevelLogByEnvironment(environment_id: number, limit
         .offset(offset)
         .limit(limit);
 
+    if (status === 'success' || status === 'failure') {
+        logs.where({ success: status === 'success' });
```
Comment (TBonnin) [inline packages/shared/lib/services/activity/activity.service.ts:226]: I don't follow the logic here.

File: packages/shared/lib/services/activity/activity.service.ts [hunk 2]
```diff
@@ -205,6 +222,36 @@ export async function getTopLevelLogByEnvironment(environment_id: number, limit
...
+    if (integration) {
+        logs.where({ provider_config_key: integration });
+    }
+
+    if (date) {
+        const dateObj = new Date(date);
+        const month = (dateObj.getUTCMonth() + 1).toString().padStart(2, '0'); // Add leading zero if needed
+        const day = dateObj.getUTCDate().toString().padStart(2, '0'); // Add leading zero if needed
+        const formattedDate = `${month}/${day}/${dateObj.getUTCFullYear()}`;
+        logs.whereRaw(`to_char(to_timestamp(_nango_activity_logs.timestamp / 1000), 'MM/DD/YYYY') = '${formattedDate}'`);
```
Comment (TBonnin) [inline packages/shared/lib/services/activity/activity.service.ts]: can we use something like this`date_trunc('day', timestamp) == ${{new Date(date.toDateString())}}`

File: packages/shared/lib/services/activity/activity.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/services/activity/activity.service.ts:226]: `success` is a column in the activity logs table

```
success: true // successful
success: false // failure
success: null // in progress
```

File: packages/shared/lib/services/activity/activity.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/activity/activity.service.ts]: Updated to use
```
logs.whereRaw("date_trunc('day', to_timestamp(_nango_activity_logs.timestamp / 1000)) = ?", [formattedDate]);
```

**Review by TBonnin — APPROVED** [review]

**Pull request merged**